packages/jsii-rosetta/lib/translate.ts

```diff
@@ -107,7 +114,7 @@ export interface SnippetTranslatorOptions extends AstRendererOptions {
 
 export interface TranslateResult {
   translation: string;
-  diagnostics: ts.Diagnostic[];
+  diagnostics: readonly ts.Diagnostic[];
```

Comment:
What does it mean to have `readonly` after the `:`?

Also mark `translation` as `readonly`?

Reply:
`readonly` here is a modifier applied to the array type... Actually means I promise (under the type checker's supervision) I will not modify the array provided as input.